### ISSUE TITLE
fix(jwt-security): published package crashes ESLint — recommended emits jwt/ prefixes after rename

### ISSUE THREAD

**Issue by ofri-peretz** [ofri-peretz/eslint#514]
**Severity: the plugin does not run at all.** Worse for an adopter than any false positive.

## Repro

```js
import jwtSecurity from 'eslint-plugin-jwt-security';
export default [{
  plugins: { 'jwt-security': jwtSecurity },
  rules: { ...jwtSecurity.configs.recommended.rules },
}];
```

```
A configuration object specifies rule "jwt/no-algorithm-none",
but could not find plugin "jwt".
```

Verified against the package published on npm. Hit while building `agents/footprint/scripts/scan-targets.ts`, which now derives plugin keys from the rules themselves to work around it.

## Root cause

The source package is still `packages/eslint-plugin-jwt`. It **publishes** as `eslint-plugin-jwt-security`, but `configs.recommended` still emits the pre-rename `jwt/` prefix. Registering under the package name — the obvious thing — fails; only registering as `jwt` works.

`postgresql-security` is suspected to have the same shape (`pg/` prefix); not yet verified, the package would not resolve locally.

## Why this is not a one-line change

Switching to `jwt-security/` **breaks every adopter currently registering as `jwt`**, which is the only registration that works today. It needs a release decision first:

- dual prefixes for a deprecation window, or
- a major with a migration note

Plus the source directory rename, the docs slug (see the docs↔package coupling constraint), and a regression lock asserting each plugin's `configs.recommended` rule prefixes match its published package name.

## Acceptance

- Registering by package name works for every published plugin
- A test iterates all plugins and asserts prefix == package suffix
